shell/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -69,45 +70,71 @@ class IPCRenderer : public gin::Wrappable<IPCRenderer> {
   const char* GetTypeName() override { return "IPCRenderer"; }
 
  private:
-  void Send(bool internal,
+  void Send(v8::Isolate* isolate,
+            bool internal,
             const std::string& channel,
-            const base::ListValue& arguments) {
-    electron_browser_ptr_->get()->Message(internal, channel, arguments.Clone());
+            v8::Local<v8::Value> arguments) {
+    blink::CloneableMessage message;
+    if (!mate::ConvertFromV8(isolate, arguments, &message)) {
+      return;
```

Comment:
In Electron 7 this will only be hit if both the V8 serializer and the `base::Value` serializer fail. I'm not exactly sure what circumstances the `base::Value` serializer can fail in, but those ones. In Electron 8 this will throw.

The `EmitDeprecationWarning` is taken care of by the `base::Value` path in the serializer.